### ISSUE TITLE
Add xml.gz support

### ISSUE THREAD

**Issue by bt4ibwem8** [blogdron/EPGTV#8]
I am now testing different EPG source, but your script does not work for it. It seems to be because of .gz compression instead of the plain .xml format. In different software it works. Here is the link: https://www.bevy.be/bevyfiles/slovakia.xml.gz

**Comment by blogdron** [comment]
`gz` supported but this URL

 * https://www.bevy.be/bevyfiles/slovakia.xml.gz

Have error `404` Not Found. Bad link.

For example 

`mpv https://smolnp.github.io/IPTVru//IPTVstable.m3u8`  


Inside `M3U` have `gnu zip` links to `EPG` archives
```
#EXTM3U url-tvg="http://epg.it999.ru/ru2.xml.gz, https://iptvx.one/epg/epg.xml.gz" tvg-shift="+3"
```

Working fine

**Comment by bt4ibwem8** [comment]
My link above is death. Here are new links:
https://open-epg.com/app/index.php

**Comment by blogdron** [comment]
`EPGTV` Work with `M3U` playlit, `EPG` `xml` or `gz` data expect inside `M3U`  

URLs to directly`EPG` files not hanles

M3U have URL links to TV streams + search ID, search ID used for find TV info in EPG. 
Random M3U can just not work with random EPG

 1  - You can add some EPG URL to you M3U playlist


For example `youTV.m3u`

```
#EXTM3U url-tvg="http://blablabla.xml, https://www.open-epg.com/files/slovakia.xml.gz"
#EXTINF:-1 tvg-id="channel_id"  group-title="Group name",channel name
http://example.com/tvstream
...blablabla
```

Now, https://www.open-epg.com/files/slovakia.xml.gz handle for you `youTV.m3u` automaticly



If with new EPG URL you no see TV info.... well 
In a normal situation, the video stream playlist and TV channel data form a pair, the playlist contains channel names and their identifiers, thanks to which you can find the channel, often those who make playlists or TV show lists call channels differently and this is a problem since it is impossible to match the video stream and TV show data, it is impossible to resolve this automatically. Sometimes TV show data fits the playlist and everything works, and sometimes it does not.

Most often, the playlist is generated for specific EPG data,
everything is adjusted to each other, and everything works.

Once again, the playlist is a separate entity, EPG data is a separate entity, if they were generated without taking each other into account, there is a high probability that they will not work with each other.

And so, EPG data in the archives is well supported, you just need to have a playlist + TV show data pair that fit each other.

I indicated above that you can insert a link with EPG data into your playlist, turn it on and see if they can work together

If the playlist contains tvg-id="CHANNEL123" and the desired channel is marked with this ID in the TV show data, then everything will work. If not, then there will be an automatic attempt to find data by the channel title or by the end of the link (but this automation is unreliable) Well, that's it. I have nothing more to say.

Archives work, but playlists may simply be incompatible with them, the archive will be downloaded, unpacked, processed, but nothing will be found in it, or maybe it will be found, depending on your luck

------------

**`xml.gz` supported and working**

**Comment by bt4ibwem8** [comment]
can you try epg service on m3u4u.com website? because .xml version works with your script but xml.gz version (of the same epg) does not work...

**Comment by blogdron** [comment]
>m3u4u.com

Not working, not open. Maybe this blocked my country

-------------

Just share you M3U file with not worked EPG url inside, Im try check it latter

------------

P.S you add path to `gzip` in config?

```lua
gzip_path = 'C:\\Program Files\\gzip\\bin\\gzip.exe', -- set fullpath to you gzip installation
```

You on windows right?

**Comment by bt4ibwem8** [comment]
yes, i am on Windows. i pasted gzip path in conf.lua but still does not work
[m3u4u.xml.gz](https://github.com/user-attachments/files/18277108/m3u4u.xml.gz)

**Comment by blogdron** [comment]
Im create test `M3U`  (just take random tv stream and set `tvg-id` from `EPG` archive )

test.m3u

```
#EXTM3U url-tvg="https://github.com/user-attachments/files/18277108/m3u4u.xml.gz"

#EXTINF:-1 tvg-id="360TuneBox(src01).cz" tvg-logo="http://epg.one/img2/675.png" group-title="Познавательные",Моя Планета (EDTV 576p)
http://194.143.148.28:8080/Nauka20/mpegts
```

test start

```
mpv test.m3u
```

all works for me, on Linux :)

![img](https://i.ibb.co/DDBxqgg/2024-12-30-22-56-02.png)



 * Maybe this EPG data no have `tvg-id` values for you `M3U` playlist, impossible fix it automaticly, need generate EPG for M3U or generate M3U for EPG. 
 * If you can, show me mpv console output. 
 * In top mpv window you see messages about EPG load/parse?

Maybe again window specific problem :(, but now im again no have windows

**Comment by bt4ibwem8** [comment]
Your test.m3u does not work for me, got the same behavior: script shows Download/Parse but then "no epg data". EPG is generated specific for my playlist (the same epg does work with plain .xml format).
I tried now also in Debian system (mpv 0.39 flatpak) and the same issue: xml works but xml.gz not... Channel ID in .gmz is the same as in the plain .xml (it is the same file, just compressed).

**Comment by blogdron** [comment]
It's strange... 🤔 . I'll try to figure it out, but after the new year. Happy New Year  🎄🥳

**Comment by bt4ibwem8** [comment]
yes, it is weird... maybe something wrong with gzip on my side.
And Happy New Year to you as well 🥳
